tests/deploy/test_scheduler_agent.py

```diff
@@ -1232,6 +1215,67 @@ def make_resources(
     assert agent.executor_manager.executors["agent2"].execute_count == 3
     assert agent.executor_manager.executors["agent3"].execute_count == 0
 
+    #########################
+    # Verify failure events #
+    #########################
+    agent.executor_manager.reset_executor_counters()
+    # make agent1's executor also managed to make r1 fail and succeed as we please
+    executor1: ManagedExecutor = ManagedExecutor()
```

Comment:
I've been very careful to avoid this in the current implementation, because it could be very confusing if any scenarios are ever added after this one. I would consider how much effort it would be to make it a separate test case. If you decide against that, I'd like an explicit comment near the top of this one, where we make executor2 a managed one, stating that later on we'll do the same for this one.

Reply:
That or revert it at the end of this scenario (e.g. with a [monkeypatch context manager](https://docs.pytest.org/en/stable/reference/reference.html#pytest.MonkeyPatch.context)), whichever you prefer.